### ISSUE TITLE
Question About Training Time

### ISSUE THREAD

**Issue by 2478452651** [mirthAI/GDM-VE#1]
Hello,

I noticed that my training time is significantly longer than what you reported in your paper. I trained the model on a single NVIDIA A100 GPU (40 GB). Since 40 GB of memory was not enough for your original batch size, I reduced it to 8. Even so, it still takes around 5.5 hours to reach 45,000 steps.

I did not modify any other parameters besides the batch size. Could you help me understand why the training is taking longer in my case?

Thank you!

**Comment by 2478452651** [comment]
Here is my training log.
[stdout.txt](https://github.com/user-attachments/files/21738894/stdout.txt)

**Comment by zhangtlzu** [comment]
Hi,
I would like to clarify that we train our model on our high-performance computing resource cluster, Hipergator (we previously had A100 80GB, but we've now replaced it with B200). It's the fastest (top 1) supercomputer owned and operated by a university in the U.S. You can see from this link:
https://ai.ufl.edu/research/hipergator/
Given this, it's normal for absolute training times to be longer than what we report. This means not only that our proposed model typically takes longer to train for the same number of iterations on other user computing resources, but also that other comparative methods, such as DDPM, also require longer training times than reported in our paper. We will briefly explain our computing resource advantages in a subsequent revision of the paper. We recommend that readers focus on relative times and the number of iterations required for convergence.

**Comment by 2478452651** [comment]
Thank you for your clarification. Since you mentioned focusing on the number of iterations, I would like to ask how you determined that the model converges after 45,000 iterations.

I have plotted a loss curve based on the training process, but I am unable to confirm whether the model converges at the 45k iteration mark. It would be helpful if you could provide some insight on how you assess convergence, especially in cases where the loss curve may not show a clear pattern.

Thank you again for your help.

<img width="939" height="564" alt="Image" src="https://github.com/user-attachments/assets/8be1db26-cecb-42f0-909f-8f4eda97272b" />

**Comment by zhangtlzu** [comment]
Hi,
You can consider your model converged when the checkpoint you saved achieves the best SSIM, which is the common approach in diffusion models. The training loss value of a diffusion model cannot reflect the model’s performance, so please do not determine convergence solely based on the training loss, you can see from this:
https://www.zhihu.com/question/623044009/answer/3228173646?share_code=zq8gUi73Q5S&utm_psn=1940687463103308702
Regarding the phenomenon of oscillating loss values, we found that for different timesteps t, the corresponding loss can vary significantly. Therefore, when sampling different t in each iteration, such as [0.1, 0.2, 0.4, 0.3] versus [0.7, 0.8, 0.9, 0.85] (assuming a batch size of 4), you may obtain completely different loss values. This is a common characteristic in diffusion models. You can refer to Figure 2 in this paper -- _Improved Denoising Diffusion Probabilistic Models_, which illustrates the variation of loss values across different t.

**Comment by 2478452651** [comment]
There is another thing that confuses me. You mentioned that training on the B200 cluster is faster than on the A100. However, in my experiments on one A100, the inference time I measured was approximately 4+ seconds faster than what was reported in your paper.

I did not modify any other parameters and sampled with your pretrained model weight. Could you please help me understand why this discrepancy occurs?

<img width="1179" height="75" alt="Image" src="https://github.com/user-attachments/assets/7080551a-c0bd-4878-bb2e-4d016e7b9d9e" />

**Comment by zhangtlzu** [comment]
Hi,

The inference time you show is per iteration (which varies with the batch size you set), while we report the inference time per patient. We will add these descriptions in the latest version of the paper. We reiterate that both training and inference time will vary depending on the user’s computational environment, including their conda setup. Since our model only requires 15 inference steps (and in our latest research we have further reduced this number, which we will disclose later), compared with DDPM’s 1000 steps, users should expect roughly a 60× speedup under the same environment. Achieving this speedup indicates that the model is functioning as expected.

**Comment by 2478452651** [comment]
Thanks for your explanation regarding the inference time. I would like to confirm whether the “inference time per patient” you mentioned corresponds to the same definition as described in the Fast-DDPM paper. In that work, the inference time is measured as the  inference time per image volume (anaverage of 360 slices), rather than the time per iteration.

<img width="439" height="190" alt="Image" src="https://github.com/user-attachments/assets/a745a04d-77e3-49d2-8fae-b40b7276d014" />

<img width="580" height="179" alt="Image" src="https://github.com/user-attachments/assets/e1ccf9d5-2028-4f8d-999c-cf7d40e24650" />

In addition, I noticed that the inference time as well as the PSNR and SSIM results reported in Fast-DDPM are consistent with those in your paper. However, in Fast-DDPM, the inference experiments were conducted on 4 A100 GPUs, while as you mentioned above, your experiments were carried out on the B200 system.Could you please clarify why the reported inference performance appears consistent across these two different hardware environments?

**Comment by zhangtlzu** [comment]
Yes, it means "per image volume".
This preprint version of the GDM paper uses the same environment as Fast-DDPM. Please refer to the officially published Fast-DDPM at this monment. As I mentioned earlier, our computing resource clusters used to have A100 GPUs, and we didn't have B200 at that time. We will release the official GDM paper later, and we will do our best to fix all confusing parts in the manuscript.
I have already emphasized several times that readers should focus on the relative acceleration factors of training and sampling rather than the absolute runtime. In any case, our model requires only a very small number of training iterations and sampling steps to achieve excellent image generation results. Regardless of the absolute time, as long as the acceleration factors of GDM compared to DDPM and Fast-DDPM are consistent, our conclusions are well supported.

**Comment by 2478452651** [comment]
Thank you very much for your detailed clarification. Your explanation has been very helpful for my understanding of this model, and it has resolved much of the confusion I previously had.

I truly appreciate your time and effort in addressing my questions, and I look forward to reading the revised version of your paper and the official release of GDM.
